src/main/appender.cpp

```diff
@@ -375,6 +411,52 @@ void Appender::FlushInternal(ColumnDataCollection &collection) {
 	context->Append(*description, collection);
 }
 
+void Appender::AppendDefaultToVector(Vector &result, idx_t column, SelectionVector &sel, idx_t count) {
+	// TODO: throw if count is > STANDARD_VECTOR_SIZE
+	if (column >= types.size()) {
+		throw InvalidInputException(
+		    "Could not append DEFAULT value of column %d, out of range index, table only has %d columns", column,
+		    types.size());
+	}
+	auto &type = types[column];
+	if (result.GetType() != type) {
+		throw InvalidInputException("Type mismatch, column has type: %s, while the provided Vector has type %s",
+		                            type.ToString(), result.GetType().ToString());
+	}
+
+	auto &executor = *expression_executor;
+
+	// This is only used to tell the ExpressionExecutor the size of our result
+	DataChunk empty_chunk;
+	empty_chunk.SetCardinality(count);
+	executor.SetChunk(empty_chunk);
+
+	// FIXME: if we are feeding the Vector to the ExecuteExpression method, we have to first make sure
+	// that the validity mask of the Vector is clean
+	context->RunFunctionInTransaction([&]() { executor.ExecuteExpression(column, result, sel); });
+}
+
+void Appender::AppendDefault() {
+	auto &default_expr = bound_defaults[column];
+	if (default_expr->type == ExpressionType::VALUE_CONSTANT) {
+		// Fast path: NULL or constant-folded
+		auto &bound_constant = default_expr->Cast<BoundConstantExpression>();
+		Append(bound_constant.value);
+		return;
+	}
+
+	auto &type = types[column];
+	Vector result(type, 1);
+	auto &executor = *expression_executor;
+	// The executor is initialized with expressions for every column, even though only some are used
+	// this makes it so that we can just use the 'column' index
+
+	// FIXME: we could optimize this further by delaying this until Flush is called, then doing it for the entire chunk
+	// once
```

Comment:
I wonder how much this actually improves, because the implementation ended up being:
execute the expression for <default_row_count> tuples, then using the SelectionVector to `tmp.GetValue(i)` into `result.SetValue(sel.get_index(i))`

Reply:
Fair point. Let's remove the `FIXME`. We can always come back to this implementation if we have reproductions for concrete slow performance/bottlenecks.